.travis.yml

```diff
@@ -0,0 +1,19 @@
+services:
+- docker
+script:
+# Ideally we'd just run compile.sh here:
+#
+# - VERBOSE=yes bash ./compile.sh
+#
+# Unfortunately bazel workers seem to hang in Travis right now so we test the
+# build from scratch in a controlled Docker container (that's... a docker
+# inside a docker inside a VM). This also helps test which packages are actually
+# required for a successful build from scratch (Travis otherwise has tons of
+# installed installed by default).
+#
+# Note: Verbosity is to avoid being killed by Travis after 10min w/o output.
+- echo "set -ex" > docker-test.sh
+- echo "apt-get update" >> docker-test.sh
+- echo "apt-get install -y automake g++ libtool make curl default-jdk git python unzip wget zip" >> docker-test.sh
```

Comment:
...and here you can another reason for the mentioned design decision: your change is pulling in quite a few new build dependencies (`automake`, `make`) that are not needed, as `protc` can be built with `bazel`.

Reply:
`protoc` can't be built with `bazel` if you don't have `bazel`, which requires `protoc` (*of the correct version*).